Userland/Libraries/LibWeb/WebAssembly/WebAssemblyTableConstructor.cpp

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2021, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Runtime/GlobalObject.h>
+#include <LibJS/Runtime/TypedArray.h>
+#include <LibWeb/Bindings/WindowObject.h>
+#include <LibWeb/WebAssembly/WebAssemblyObject.h>
+#include <LibWeb/WebAssembly/WebAssemblyTableConstructor.h>
+#include <LibWeb/WebAssembly/WebAssemblyTableObject.h>
+#include <LibWeb/WebAssembly/WebAssemblyTablePrototype.h>
+
+namespace Web::Bindings {
+
+WebAssemblyTableConstructor::WebAssemblyTableConstructor(JS::GlobalObject& global_object)
+    : NativeFunction(*global_object.function_prototype())
+{
+}
+
+WebAssemblyTableConstructor::~WebAssemblyTableConstructor()
+{
+}
+
+JS::Value WebAssemblyTableConstructor::call()
+{
+    vm().throw_exception<JS::TypeError>(global_object(), JS::ErrorType::ConstructorWithoutNew, "WebAssembly.Table");
+    return {};
+}
+
+JS::Value WebAssemblyTableConstructor::construct(FunctionObject&)
+{
+    auto& vm = this->vm();
+    auto& global_object = this->global_object();
+
+    auto descriptor = vm.argument(0).to_object(global_object);
+    if (vm.exception())
+        return {};
+
+    auto element_value = descriptor->get("element");
+    if (vm.exception())
+        return {};
+    if (!element_value.is_string()) {
+        vm.throw_exception<JS::TypeError>(global_object, JS::ErrorType::InvalidHint, element_value.to_string_without_side_effects());
+        return {};
+    }
+    auto& element = element_value.as_string().string();
+
+    Optional<Wasm::ValueType> reference_type;
+    if (element == "anyfunc"sv)
+        reference_type = Wasm::ValueType(Wasm::ValueType::FunctionReference);
+    else if (element == "externref"sv)
+        reference_type = Wasm::ValueType(Wasm::ValueType::ExternReference);
+
+    if (!reference_type.has_value()) {
+        vm.throw_exception<JS::TypeError>(global_object, JS::ErrorType::InvalidHint, element);
+        return {};
+    }
+
+    auto initial_value = descriptor->get("initial");
+    if (vm.exception())
+        return {};
+    auto maximum_value = descriptor->get("maximum");
+    if (vm.exception())
+        return {};
+
+    if (initial_value.is_empty()) {
+        vm.throw_exception<JS::TypeError>(global_object, JS::ErrorType::NotA, "Number");
+        return {};
+    }
+
```

Comment:
`get()` no longer returns empty values in non-error cases, so this never applies

```suggestion
```

Reply:
I stop looking at LibJS for like a month and then half the APIs have changed
neat :P